packages/odyssey-react-mui/src/labs/DataView/TableLayoutContent.tsx

```diff
@@ -321,7 +316,7 @@ const TableLayoutContent = <TData extends MRT_RowData>({
   const hasColumnWithGrow = columns.some((column) => column.grow === true);
 
   const dataTable = useMaterialReactTable<TData>({
-    data: (!isEmpty && !isNoResults ? data : []) as TData[],
+    data: !isEmpty && !isNoResults ? data : [],
```

Comment:
Interesting. Did we need to remove the `as TData[]`? Usually you wanna say `never[]` or `any[]` to be `TData[]`:

```suggestion
    data: !isEmpty && !isNoResults ? data : ([] as TData[]),
```

TS can't infer the type of `[]` without an `as` unless you add at least one item to it.

I wonder if we can use `satisfies` instead:
```suggestion
    data: !isEmpty && !isNoResults ? data : ([] satisfies TData[]),
```